### ISSUE TITLE
i2c error

### ISSUE THREAD

**Issue by Isioviel** [KitronikLtd/micropython-microbit-kitronik-klef-piano#1]
Hi, I am enjoying coding my piano, however am getting frequent I2C error messages - it's not always on the same line, sometimes read and sometimes write. Sometimes it happens immediately / without me touching any keys, sometimes I can play notes for a while before it will crash.

Example error message:
OSError I2C write  error - 1010

Could you help? It is working fine using your MakeCode files, however I would love to get it working with micropython instead!

**Comment by AlasdairAtKitronik** [comment]
@Isioviel Thanks for raising the issue, and for coming and speaking to me at BETT. I've managed to replicate the issue here and am starting work on trying to find a fix. Initial thoughts are that the capacative touch chip is entering a power saving sleep mode which is stopping the I2C communication.

**Comment by AlasdairAtKitronik** [comment]
Issue now fixed with commit:
c8b43b8957e485cf2e02eeadf912b569ddfa6fee

**Comment by kevinjwalters** [comment]
I get this too with the latest code, from Chrome's webserial connection:

```
Traceback (most recent call last):
  File "__main__", line 127, in <module>
  File "__main__", line 42, in __init__
OSError: I2C read error -1010
```

I have two usb connectors connected at the moment to allow me to do that...

I just tried MakeCode with the extension for klef and that worked first time so the hardware is ok.

**Comment by kevinjwalters** [comment]
I had a look at that last night and eventually spotted the problem. I've offered a fix for it in #3
